tests/ci/run_cryptofuzz.sh

```diff
@@ -11,50 +11,56 @@ set -u
 
 rm -rf "$BUILD_ROOT"
 mkdir -p "$BUILD_ROOT"
-cd "$BUILD_ROOT"
 
-# Build AWS-LC based on https://github.com/guidovranken/cryptofuzz/blob/master/docs/openssl.md
-${CMAKE_COMMAND} -DCMAKE_CXX_FLAGS="$CXXFLAGS" -DCMAKE_C_FLAGS="$CFLAGS" -DBORINGSSL_ALLOW_CXX_RUNTIME=1 \
-  -GNinja -DBUILD_TESTING=OFF -DBUILD_LIBSSL=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo ../
-ninja
-cd ../
+# Build ACCP, but skip the tests since that's covered by all the other CI
+cd ${SRC_ROOT}
+./gradlew -x check build
+AWSLC_INSTALL_PATH=`realpath build/awslc/bin/`
+export AWSLC_INCLUDE_PATH="${AWSLC_INSTALL_PATH}/include/"
+export AWSLC_LIBRARY_PATH="${AWSLC_INSTALL_PATH}/lib64/"
+export CXXFLAGS="$CXXFLAGS -I $AWSLC_INCLUDE_PATH"
+export ACCP_JAR=`realpath build/lib/AmazonCorrettoCryptoProvider.jar`
 
-# CRYPTOFUZZ_AWSLC will enable AES_256_XTS in upstream code. See `V610453347`.
-# TODO: remove CRYPTOFUZZ_BORINGSSL when CRYPTOFUZZ_AWSLC is fully integrated.
-export CXXFLAGS="$CXXFLAGS -DCRYPTOFUZZ_BORINGSSL -DCRYPTOFUZZ_AWSLC"
-export OPENSSL_INCLUDE_PATH=`realpath include/`
-export OPENSSL_LIBCRYPTO_A_PATH=`realpath ${BUILD_ROOT}/crypto/libcrypto.a`
+# We run the Java module with ACCP as the JCE provider.
+# This module uses OpenSSL symbols, so we provide those from AWS-LC to avoid symbol conflicts
+# Java https://github.com/guidovranken/cryptofuzz/blob/master/docs/java.md
+export JDK_PATH=${JAVA_HOME}
+export LINK_FLAGS="${LINK_FLAGS} -L${JDK_PATH}/lib/server/ -ljvm -L${AWSLC_LIBRARY_PATH} -lcrypto"
+export LINK_FLAGS="${LINK_FLAGS} -Wl,-rpath=${JDK_PATH}/lib/server/:${AWSLC_LIBRARY_PATH}"
+export CXXFLAGS="$CXXFLAGS -DCRYPTOFUZZ_JAVA -DJAVA_WITH_ECDSA -DCRYPTOFUZZ_ACCP"
+cd "${CRYPTOFUZZ_SRC}/modules/java/"
+make -j$(nproc)
+
+# LSAN suppressions due to false positives
+cd ${FUZZ_ROOT}
+echo 'leak:libjvm.so' >> lsan.supp
+echo 'leak:libz.so' >> lsan.supp
+echo 'leak:libzip.so' >> lsan.supp
```

Comment:
i understand why libjvm would trigger false positives for LSAN, but why would compression libraries do so?

Reply:
These libraries are invoked by libjvm and they get flagged by LSAN if you don't suppress them. My theory is that libjvm's leakiness gets picked up by its dependencies because of similar issues around how it deals with memory.